server/src/main/scala/za/co/absa/atum/web/api/provider/PostgresAccessProvider.scala

```diff
@@ -14,38 +14,29 @@
  * limitations under the License.
  */
 
-package za.co.absa.atum.web.api.service
+package za.co.absa.atum.web.api.provider
 
-import com.typesafe.config.{Config, ConfigFactory}
-import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.stereotype.Component
-import slick.jdbc.JdbcBackend.Database
+import com.typesafe.config.{Config, ConfigValueFactory}
+import za.co.absa.atum.web.api.AtumConfig
 import za.co.absa.atum.web.api.database.Runs
-import za.co.absa.atum.web.api.service.utils.ExecutorsProvider
+import za.co.absa.fadb.slick.FaDbPostgresProfile.api._
 import za.co.absa.fadb.slick.SlickPgEngine
 
-/**
- *
- * @param executors
- */
-@Component
-class PostgresAccessProvider@Autowired()( executors: ExecutorsProvider ) {
-
-  private val config = ConfigFactory.load("application.properties")
-
-  /**
-   *
-   * @return
-   */
-  private def dbConfig: Config = {
-    val conf = config.getConfig("postgres")
-    print("Configs:", conf)
-    conf
+import scala.concurrent.ExecutionContext
+
+class PostgresAccessProvider {
+
+  val executor: ExecutionContext = ExecutionContext.Implicits.global
+
+  private def databaseConfig: Config = {
+    val baseConfig = AtumConfig.dbConfig
+    Map.empty.foldLeft(baseConfig) { case (acc, (configPath, configVal)) =>
```

Comment:
What's the point of this code?
It's a `foldLeft` on an empty Map, so no op.

Reply:
I added a TODO comment as promised